qiskit_nature/algorithms/ground_state_solvers/adapt_vqe.py

```diff
@@ -40,15 +42,28 @@
 
 
 class AdaptVQE(GroundStateEigensolver):
-    """A ground state calculation employing the AdaptVQE algorithm."""
-
+    """A ground state calculation employing the AdaptVQE algorithm.
+
+    The performance of AdaptVQE can significantly depend on the choice of gradient method, QFI
+    solver (if applicable) and the epsilon value.
+    """
+
+    @deprecate_arguments(
+        "0.4.0",
+        {"delta": "gradient"},
+        additional_msg=(
+            "Instead of `delta=1.0` you have to construct a gradient, like so "
+            "`gradient=Gradient(grad_method='fin_diff', epsilon=1.0)`."
+        ),
+    )
```

Comment:
Normally when deprecating something the code should continue to work as it did. E.g if I had code that used/set delta it ought to work here albeit we see some deprecation message saying that in the future only gradient will be supported. Hence I would expect to see paths in the code for if delta is used (that will be removed at some point) and paths for Gradient. As far as I can see delta is set on the instance but otherwise ignored; a default gradient is created if your code uses it that  way and predates the gradient being added, the code then just works on this gradient.